src/replication.c

```diff
@@ -2859,9 +2859,11 @@ void dualChannelSyncSuccess(void) {
     server.primary_initial_offset = server.repl_provisional_primary.reploff;
     replicationResurrectProvisionalPrimary();
     /* Wait for the accumulated buffer to be processed before reading any more replication updates */
-    if (streamReplDataBufToDb(server.primary) == C_ERR) {
+    if (server.pending_repl_data.blocks && streamReplDataBufToDb(server.primary) == C_ERR) {
         /* Sync session aborted during repl data streaming. */
         serverLog(LL_WARNING, "Failed to stream local replication buffer into memory");
+        replicationAbortDualChannelSyncTransfer();
+        replicationUnsetPrimary();
```

Comment:
This crashed in the sanitizer test: https://github.com/valkey-io/valkey/actions/runs/10066315835/job/27827425020?pr=804